src/uct/base/uct_iface.c

```diff
@@ -242,6 +242,27 @@ static int uct_iface_is_same_device(const uct_iface_h iface,
     return !memcmp(device_addr, dev_addr, attr.device_addr_len);
 }
 
+void uct_iface_populate_info_str_buf(
+                                const uct_iface_is_reachable_params_t *params,
+                                const char *fmt, ...)
+{
+    char *info_str_buf = UCS_PARAM_VALUE(UCT_IFACE_IS_REACHABLE_FIELD,
+                                    params, info_string, INFO_STRING,
+                                    NULL);
+    size_t info_str_buf_len = UCS_PARAM_VALUE(UCT_IFACE_IS_REACHABLE_FIELD,
+                                        params, info_string_length,
+                                        INFO_STRING_LENGTH, 0);
+
+    va_list args;
+    va_start(args, fmt);
+
+    if (info_str_buf && info_str_buf_len > 0 && info_str_buf[0] == '\0') {
+        snprintf(info_str_buf, info_str_buf_len, fmt, args);
```

Comment:
`ucs_snprintf_safe`

Reply:
AFAIU, it might be called several times in a single flow (e.g `uct_ib_iface_dev_addr_is_reachable` then in the caller `uct_ib_iface_is_reachable_v2`). The strings overwrite each other?

Reply:
@gleon99  This is why I am checking `info_str_buf` is empty (`info_str_buf[0] == '\0'`), so it won't override an existing reason. I assume the first time `info_str_buf` is filled happens in the deepest level of this check (== the most descriptive one), yet I want to provide a more basic reason in case it wasn't filled earlier in the flow.

@yosefe it answers your other question about why we need `info_str_buf[0] == '\0'` .